### PR TITLE
fix(forms): parse exponential notation in numberInputType parser

### LINKED ISSUE #12121: exponential notation in input[type=number] validation
If we provide some long number, like 111111111111111111111111 in the input[type=number] and then click this little up/down arrow, then input value is converted to exponential notation: 1.11111111111111111e+23

Angular consider this is invalid value, but i think it should be still valid:
```
> typeof 1.11111111111111111e+23
< "number"
```

See this plunker:
http://embed.plnkr.co/hBmk7d/preview

### PR DESCRIPTION
**Pull request by caitp**
Closes #12121

### CODE REVIEW

File: src/ng/directive/input.js [hunk 1]
```diff
@@ -13,7 +13,7 @@
 var ISO_DATE_REGEXP = /\d{4}-[01]\d-[0-3]\dT[0-2]\d:[0-5]\d:[0-5]\d\.\d+([+-][0-2]\d:[0-5]\d|Z)/;
 var URL_REGEXP = /^(ftp|http|https):\/\/(\w+:{0,1}\w*@)?(\S+)(:[0-9]+)?(\/|\/([\w#!:.?+=&%@!\-\/]))?$/;
 var EMAIL_REGEXP = /^[a-z0-9!#$%&'*+\/=?^_`{|}~.-]+@[a-z0-9]([a-z0-9-]*[a-z0-9])?(\.[a-z0-9]([a-z0-9-]*[a-z0-9])?)*$/i;
-var NUMBER_REGEXP = /^\s*(\-|\+)?(\d+|(\d*(\.\d*)))\s*$/;
+var NUMBER_REGEXP = /^\s*(\-|\+)?(\d+|(\d*(\.\d*)))([eE]\+?\d+)?\s*$/;
```
Comment (lgalfaso) [inline src/ng/directive/input.js]: No negative exponents support?

Comment (caitp) [inline src/ng/directive/input.js]: ah good point, amending

**Comment by lgalfaso** [comment]
LGTM

**Comment by caitp** [comment]
thx

**Comment by Narretz** [comment]
Does this fix https://github.com/angular/angular.js/issues/11602 too?

**Comment by gkalpak** [comment]
Trying the [plnkr with snapshot][1] confirms it does.

[1]: http://plnkr.co/edit/wK0Og8zHiqdHjcGCmA0A?p=preview

**Comment by Narretz** [comment]
Thx @gkalpak

**Pull request closed without merging**